File: android/src/main/java/dev/barooni/capacitor/calendar/models/inputs/CheckPermissionInput.kt
package dev.barooni.capacitor.calendar.models.inputs

import com.getcapacitor.PluginCall
import dev.barooni.capacitor.calendar.PluginError
import dev.barooni.capacitor.calendar.models.enums.CalendarPermissionScope

data class CheckPermissionInput(
    private val call: PluginCall,
) {
    val scope: CalendarPermissionScope

    init {
        val rawScope = call.getString("scope") ?: throw PluginError.MissingScope
        scope = CalendarPermissionScope.fromValue(rawScope) ?: throw PluginError.InvalidScope

        if (scope == CalendarPermissionScope.WRITE_REMINDERS || scope == CalendarPermissionScope.READ_REMINDERS) {
            throw PluginError.InvalidScope
        }
    }
}

File: ios/Plugin/Models/Inputs/CheckPermissionInput.swift
import Capacitor

struct CheckPermissionInput {
    private let scope: CalendarPermissionScope

    init(call: CAPPluginCall) throws {
        guard let rawScope = call.getString("scope") else {
            throw PluginError.scopeMissing
        }
        guard let scope = CalendarPermissionScope(rawValue: rawScope) else {
            throw PluginError.invalidScope
        }
        self.scope = scope
    }

    func getScope() -> CalendarPermissionScope {
        return scope
    }
}
